### PR TITLE
DS-236: Add support for images in action blocks (and other cleanup)

### PR DESCRIPTION
**Pull request by remydenton**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-236

## Summary

Allows images and other freeform content to be passed as content in action block items

## Details

- Removes the `icon` property (with deprecation) and adds a new `content` property.   There was an unused `__icon` class that was also removed.  Now, instead of passing an object for the icon, you can pass an already rendered icon or anything else for content.
- Renames `contentItems` to `items` (with deprecation)
- Adds support for passing an item without a url.  This makes something like the partner links shown at https://boltdesignsystem.com/pattern-lab/?p=pages-event-detail possible.
- Updates the partner links to require less custom CSS and delegate more to action blocks.
- Replaces the private _action-blocks-item.twig file with a public action-block.twig file.  This was based on a discussion with @danielamorse about architecture for patterns with multiple items.

## How to test

- Confirm that images and other content can now be passed to action blocks as shown in the new example at /pattern-lab/?p=components-action-blocks-freeform-content
- Confirm that action blocks examples shown at the following URLs are unchanged.
/pattern-lab/patterns/40-components-action-blocks/index.html
/pattern-lab/?p=pages-event-detail

### CODE REVIEW

**Review by mikemai2awesome — CHANGES_REQUESTED** [review]
Can we rename icon to media instead? We can rename text to content later. Don't rename icon to content.

**Review by danielamorse — APPROVED** [review]
@remydenton updated `content` prop to `media` as discussed. Left `text` alone for now.

Approved. There is a slight layout change on [this demo](https://boltdesignsystem.com/pattern-lab/?p=components-action-blocks-spacing-variations), but it's expected because the icon sizes changed.

**Review by mikemai2awesome — APPROVED** [review]

File: docs-site/src/pages/pattern-lab/_patterns/10-visual-styles/20-information-density/_info-density.twig [hunk 1]
```diff
@@ -145,49 +145,79 @@
...
     {% include "@bolt-components-action-blocks/action-blocks.twig" with {
-      contentItems: [
+      items: [
         {
           text: "Item 1 with a long line of text",
           url: "#!",
-          icon: {
-            name: "download",
-          }
+          content: icon_1,
```
Comment (mikemai2awesome) [inline docs-site/src/pages/pattern-lab/_patterns/10-visual-styles/20-information-density/_info-density.twig:195]: This along with other docs still need to be updated to `media`. It sucks we don't have validation now that we are not aware of errors. @danielamorse @remydenton

**Review by mikemai2awesome — CHANGES_REQUESTED** [review]
Oops, just caught a lot of other docs didn't get updated to `media`. See inline comment.

File: docs-site/src/pages/pattern-lab/_patterns/10-visual-styles/20-information-density/_info-density.twig (hunk 1 above)
Comment (danielamorse) [reply docs-site/src/pages/pattern-lab/_patterns/10-visual-styles/20-information-density/_info-density.twig]: The rest have been updated @mikemai2awesome: https://github.com/boltdesignsystem/bolt/pull/1995/commits/7c92d7ed8e0a8498ceb681966ce5730f59460ced

**Review by mikemai2awesome — APPROVED** [review]

**Pull request merged**